code/backend/pages/03_Delete_Data.py

```diff
@@ -35,59 +39,44 @@
 # Inject CSS with Markdown
 st.markdown(hide_table_row_index, unsafe_allow_html=True)
 
+try:
+    if env_helper.AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION:
+        search_handler: IntegratedVectorizationSearchHandler = (
+            IntegratedVectorizationSearchHandler(env_helper)
+        )
+        search_client = search_handler.search_client
+    else:
+        search_handler: AzureSearchHandler = AzureSearchHandler(env_helper)
+        search_client = search_handler.search_client
 
-def get_files():
-    return search_client.search("*", select="id, title", include_total_count=True)
-
-
-def output_results(results):
-    files = {}
+    results = search_handler.get_files()
     if results.get_count() == 0:
         st.info("No files to delete")
         st.stop()
     else:
         st.write("Select files to delete:")
 
-    for result in results:
-        id = result["id"]
-        filename = result["title"]
-        if filename in files:
-            files[filename].append(id)
-        else:
-            files[filename] = [id]
-            st.checkbox(filename, False, key=filename)
-
-    return files
-
-
-def delete_files(files):
-    ids_to_delete = []
-    files_to_delete = []
-
-    for filename, ids in files.items():
-        if st.session_state[filename]:
-            files_to_delete.append(filename)
-            ids_to_delete += [{"id": id} for id in ids]
-
-    if len(ids_to_delete) == 0:
-        st.info("No files selected")
-        st.stop()
-
-    search_client.delete_documents(ids_to_delete)
-
-    st.success("Deleted files: " + str(files_to_delete))
-
-
-try:
-    vector_store_helper: AzureSearchHelper = AzureSearchHelper()
-    search_client = vector_store_helper.get_vector_store().client
-
-    results = get_files()
-    files = output_results(results)
+    files = search_handler.output_results(results)
+    selections = {
+        filename: st.checkbox(filename, False, key=filename)
+        for filename in files.keys()
+    }
+    selected_files = {
```

Comment:
Does this get updated as the user selects each item? Asking as it looks like this code is only invoked on loading, but it might be some streamlit magic making it work! 😅

Reply:
Streamlit magic 🪄  It checks on every checkbox click